src/lucky/routeable.cr

```diff
@@ -115,4 +145,81 @@ module Lucky::Routeable
       \{% raise "Use `route` instead of `with` if the action doesn't need params" %}
     end
   end
+
+  macro included
+    PARAM_DECLARATIONS = [] of Crystal::Macros::TypeDeclaration
+
+    macro inherited
+      PARAM_DECLARATIONS = [] of Crystal::Macros::TypeDeclaration
+    end
+  end
+
+  macro param(type_declaration)
+    {% PARAM_DECLARATIONS << type_declaration %}
+
+    def {{ type_declaration.var }} : {{ type_declaration.type }}
+      val = params.get(:{{ type_declaration.var.id }})
+      if val.nil?
+        nil || {{ type_declaration.value || nil }}
+      else
+        {% if type_declaration.type.class_name == "Union" %}
+          result = {{ type_declaration.type.types.first }}::LuckyHack.parse(val)
+          if result.is_a? {{ type_declaration.type.types.first }}::LuckyHack::SuccessfulCast
+            result.value
+          else
+            nil
+          end
+        {% else %}
+          result = {{ type_declaration.type }}::LuckyHack.parse(val)
+          if result.is_a? {{ type_declaration.type }}::LuckyHack::SuccessfulCast
```

Comment:
We could remove that check if `FailedCast` would have `def value; nil; end` right?